packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -1221,6 +1230,42 @@ function updateTransition(
   return [start, isPending];
 }
 
+let clientId: number = 0;
+
+function mountOpaqueReference(): string | IdObject {
+  const [id, setId] = mountState(() => {
+    if (getIsHydrating()) {
+      return {
+        $$typeof: REACT_OPAQUE_OBJECT_TYPE,
+        _setId: () => {
```

Comment:
It is possible for this function to be called more than once if the object is referenced multiple times. This would eventually get rendered but unnecessarily queue up many updates and waste IDs that won't get used. We might want to dedupe so that if it's called more than once, we don't generate a new id each time.